### PR TITLE
ACS-3793 Permissions tests for creating rules.

### PR DESCRIPTION
**Pull request by tpage-alfresco**

### CODE REVIEW

**Review by KikoUK — APPROVED** [review]

**Review by gevagel — APPROVED** [review]

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java [hunk 1]
```diff
@@ -370,6 +371,97 @@ public RestRuleModel testRolePermissionsWith(UserRole userRole)
...
+
+        STEP("Check admin can create a rule.");
+        RestRuleModel ruleModel = rulesUtils.createRuleModelWithDefaultValues();
+        restClient.authenticateUser(dataUser.getAdminUser()).withPrivateAPI().usingNode(folder).usingDefaultRuleSet().createSingleRule(ruleModel);
+
+        restClient.assertStatusCodeIs(CREATED);
+    }
+
+    /** Check that a coordinator can create rules in folders outside sites. */
+    @Test
```
Comment (mpichura) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java:409]:
```suggestion
    @Test (groups = { TestGroup.REST_API, TestGroup.RULES })
```

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java [hunk 2]
```diff
@@ -370,6 +371,97 @@ public RestRuleModel testRolePermissionsWith(UserRole userRole)
...
+
+        STEP("Check the coordinator can create a rule.");
+        RestRuleModel ruleModel = rulesUtils.createRuleModelWithDefaultValues();
+        restClient.authenticateUser(coordinator).withPrivateAPI().usingNode(folder).usingDefaultRuleSet().createSingleRule(ruleModel);
+
+        restClient.assertStatusCodeIs(CREATED);
+    }
+
+    /** Check that an editor cannot create rules in folders outside sites. */
+    @Test
```
Comment (mpichura) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java:428]:
```suggestion
    @Test (groups = { TestGroup.REST_API, TestGroup.RULES })
```

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java [hunk 3]
```diff
@@ -370,6 +371,97 @@ public RestRuleModel testRolePermissionsWith(UserRole userRole)
...
+
+        STEP("Check the editor can create a rule.");
+        RestRuleModel ruleModel = rulesUtils.createRuleModelWithDefaultValues();
+        restClient.authenticateUser(editor).withPrivateAPI().usingNode(folder).usingDefaultRuleSet().createSingleRule(ruleModel);
+
+        restClient.assertStatusCodeIs(FORBIDDEN);
+    }
+
+    /** Check that a collaborator cannot create rules in folders outside sites. */
+    @Test
```
Comment (mpichura) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java:447]:
```suggestion
    @Test (groups = { TestGroup.REST_API, TestGroup.RULES })
```

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java [hunk 4]
```diff
@@ -188,7 +189,7 @@ public void requireReadPermissionToCreateRule()
         restClient.assertLastError().containsSummary("Insufficient permissions to manage rules");
     }
 
-    /** Check that a Collaborator cannot create a rule in a private folder. */
+    /** Check that a Collaborator cannot create a rule in a folder in a private site. */
     public void siteCollaboratorCannotCreateRule()
```
Comment (mpichura) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java:193]: Do we need `@Test (groups = { TestGroup.REST_API, TestGroup.RULES })` annotation for the method to be executed?

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java [hunk 5]
```diff
@@ -206,7 +207,7 @@ public void siteContributorCannotCreateRule()
         restClient.assertLastError().containsSummary("Insufficient permissions to manage rules");
     }
 
-    /** Check that a Consumer cannot create a rule in a private folder. */
+    /** Check that a Consumer cannot create a rule in a folder in a private site. */
     public void siteConsumerCannotCreateRule()
```
Comment (mpichura) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java:211]: Do we need `@Test (groups = { TestGroup.REST_API, TestGroup.RULES })` annotation for the method to be executed?

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java [hunk 6]
```diff
@@ -215,7 +216,7 @@ public void siteConsumerCannotCreateRule()
         restClient.assertLastError().containsSummary("Insufficient permissions to manage rules");
     }
 
-    /** Check that a siteManager can create a rule in a private folder. */
+    /** Check that a siteManager can create a rule in a folder in a private site. */
     public void siteManagerCanCreateRule()
```
Comment (mpichura) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java:220]: Do we need `@Test (groups = { TestGroup.REST_API, TestGroup.RULES })` annotation for the method to be executed?

**Review by mpichura — APPROVED** [review]
LGTM.
See my comments/suggestions regarding `@Test (groups = { TestGroup.REST_API, TestGroup.RULES })`  annotation

**Review by krdabrowski — APPROVED** [review]

**Pull request merged**